src/Microsoft.Health.Dicom.Functions/IndexMetricsCollection/IndexMetricsCollectionFunction.cs

```diff
@@ -0,0 +1,80 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Diagnostics;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Core;
+using Microsoft.Health.Dicom.Core.Configs;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Features.Store;
+using Microsoft.Health.Dicom.Functions.IndexMetricsCollection.Telemetry;
+
+namespace Microsoft.Health.Dicom.Functions.IndexMetricsCollection;
+
+/// <summary>
+///  A function for collecting index metrics
+/// </summary>
+public class IndexMetricsCollectionFunction
+{
+    private readonly IIndexDataStore _indexDataStore;
+    private readonly IndexMetricsCollectionMeter _meter;
+    private readonly bool _externalStoreEnabled;
+    private readonly bool _enableDataPartitions;
+
+    public IndexMetricsCollectionFunction(
+        IIndexDataStore indexDataStore,
+        IOptions<FeatureConfiguration> featureConfiguration,
+        IndexMetricsCollectionMeter meter)
+    {
+        EnsureArg.IsNotNull(featureConfiguration, nameof(featureConfiguration));
+        _indexDataStore = EnsureArg.IsNotNull(indexDataStore, nameof(indexDataStore));
+        _meter = EnsureArg.IsNotNull(meter, nameof(meter));
+        _externalStoreEnabled = featureConfiguration.Value.EnableExternalStore;
+        _enableDataPartitions = featureConfiguration.Value.EnableDataPartitions;
+    }
+
+
+    /// <summary>
+    /// Asynchronously collects index metrics.
+    /// </summary>
+    /// <param name="invocationTimer">The timer which tracks the invocation schedule.</param>
+    /// <param name="log">A diagnostic logger.</param>
+    /// <returns>A task that represents the asynchronous metrics collection operation.</returns>
+    [FunctionName(nameof(IndexMetricsCollectionFunction))]
+    public async Task Run(
+        [TimerTrigger(IndexMetricsCollectionOptions.Frequency)] TimerInfo invocationTimer,
+        ILogger log)
+    {
+        EnsureArg.IsNotNull(invocationTimer, nameof(invocationTimer));
+        EnsureArg.IsNotNull(log, nameof(log));
+
+        log.LogInformation("Collecting a daily summation starting. At: {Timestamp}", Clock.UtcNow);
```

Comment:
Do not use `Clock`. Use `DateTime.UtcNow` for the current time. You may also want to log the time you're running for found in `TimerInfo` (I think).